src/catalog/postgres/pg_statistic_impl.cpp

```diff
@@ -129,4 +130,137 @@ bool PgStatisticImpl::DeleteColumnStatistics(const common::ManagedPointer<transa
+
+  switch (type) {
+    case type::TypeId::BOOLEAN:
+      return CreateColumnStats<execution::sql::BoolVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                        top_k_str, histogram_str, type);
+    case type::TypeId::TINYINT:
+    case type::TypeId::SMALLINT:
+    case type::TypeId::INTEGER:
+    case type::TypeId::BIGINT:
+      return CreateColumnStats<execution::sql::Integer>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                        top_k_str, histogram_str, type);
+    case type::TypeId::REAL:
+      return CreateColumnStats<execution::sql::Real>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                     top_k_str, histogram_str, type);
+    case type::TypeId::DECIMAL:
+      return CreateColumnStats<execution::sql::DecimalVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                           top_k_str, histogram_str, type);
+    case type::TypeId::TIMESTAMP:
+      return CreateColumnStats<execution::sql::TimestampVal>(table_oid, col_oid, num_rows, non_null_rows,
+                                                             distinct_values, top_k_str, histogram_str, type);
+    case type::TypeId::DATE:
+      return CreateColumnStats<execution::sql::DateVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                        top_k_str, histogram_str, type);
+    case type::TypeId::VARCHAR:
+    case type::TypeId::VARBINARY:
+      /*
+       * TODO(Joe Koshakow) The current implementation of Histogram produces meaningless results for strings. See
+       * histogram.h for a more detailed explanation. For now we just use an empty histogram.
+       */
+      return CreateColumnStats<execution::sql::StringVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                          top_k_str, nullptr, type);
+    default:
+      UNREACHABLE("Invalid column type");
+  }
+}
+
+template <typename T>
+std::unique_ptr<optimizer::ColumnStatsBase> PgStatisticImpl::CreateColumnStats(
+    table_oid_t table_oid, col_oid_t col_oid, size_t num_rows, size_t non_null_rows, size_t distinct_values,
+    const storage::VarlenEntry *top_k_str, const storage::VarlenEntry *histogram_str, type::TypeId type) {
+  using CppType = decltype(T::val_);
+  auto top_k = top_k_str != nullptr
+                   ? optimizer::TopKElements<CppType>::Deserialize(top_k_str->Content(), top_k_str->Size())
+                   : optimizer::TopKElements<CppType>();
+  auto histogram = histogram_str != nullptr
+                       ? optimizer::Histogram<CppType>::Deserialize(histogram_str->Content(), histogram_str->Size())
+                       : optimizer::Histogram<CppType>();
```

Comment:
When we first create a table all rows in pg_statistic for that table will have `NULL` in the histogram and topk columns. If you `ANALYZE` an empty table I'm actually not sure if the histogram and topk columns will be `NULL` or just a serialized empty data structure. Either way we get an empty top_k and histogram when the table is empty. The selectivity/cardinality calculations will handle this case because they just immediately return 0 on an empty table and never look at either data structure.

I'll add a comment in the code briefly explaining this.